### ISSUE TITLE
Modify the code to work on Jessie (python 3.4.2 and context argument)

### ISSUE THREAD

**Issue by julianharty** [kiwix/kiwix-build#31]
The default python3 on Raspbian Jessie is 3.4.2 

The current code uses 'context' which is first available in Python 3.4.3 

Let's see if we can find a way to work on 3.4.2 so kiwix-build can run on Raspbian Jessie.

**Comment by kelson42** [comment]
The general approach is to cross-compile things. If we do that here too, then the problem should be fixed.

**Comment by julianharty** [comment]
As a note, here's the python 3 note for the relevant change http://bugs.python.org/issue22366

**Comment by julianharty** [comment]
and this seems to be their change to implement the context https://hg.python.org/cpython/rev/13f46fc1a002

**Comment by julianharty** [comment]
@kelson42 FYI Mattieu asked for this issue to be raised in an email discussion with mossroy last week, which is why I've done so. For me, I'll probably try running a later version of python3 on my Pi, at least for now.

**Comment by mgautierfr** [comment]
@julianharty You will spend a long time compiling on a Pi directly.
You can try the PR #30 who add cross-compilation for raspberryPI. 

From the request itself (make the code to work on Raspbian Jessie), we know we need to fix #27 but it may have other thing to fix. I would have prefer to keep #27 opened as it is more self-explanatory that this one.

**Comment by julianharty** [comment]
@mgautierfr I'm trying PR https://github.com/kiwix/kiwix-build/pull/30 directly on a Pi 3 (I have several Pi's available for these experiments). Perhaps one of the needed packages isn't available from the current source (I don't quite understand where `xapian-core` being downloaded from or why the download is being skipped - I'll investigate further, however any suggestions of where to look would help) 
```
prepare sources xapian-core :
  download xapian-core : SKIP
  extract xapian-core : ERROR
Traceback (most recent call last):
  File "/home/pi/kiwix-dev/kiwix-build/utils.py", line 75, in extract_archive
    archive = tarfile.open(archive_path)
  File "/home/pi/.pyenv/versions/3.4.3/lib/python3.4/tarfile.py", line 1553, in open
    raise ReadError("file could not be opened successfully")
tarfile.ReadError: file could not be opened successfully

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "kiwix-build.py", line 862, in <module>
    builder.run()
  File "kiwix-build.py", line 831, in run
    self.prepare_sources()
  File "kiwix-build.py", line 812, in prepare_sources
    source.prepare()
  File "/home/pi/kiwix-dev/kiwix-build/dependency_utils.py", line 117, in prepare
    self.command('extract', self._extract)
  File "/home/pi/kiwix-dev/kiwix-build/dependency_utils.py", line 86, in command
    return self.target.command(*args, **kwargs)
  File "/home/pi/kiwix-dev/kiwix-build/dependency_utils.py", line 50, in command
    ret = function(*args, context=context)
  File "/home/pi/kiwix-dev/kiwix-build/dependency_utils.py", line 106, in _extract
    name=self.source_dir)
  File "/home/pi/kiwix-dev/kiwix-build/utils.py", line 115, in extract_archive
    if archive is not None:
UnboundLocalError: local variable 'archive' referenced before assignment

```
